### PR TITLE
Remove turkish specialization.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @jasonmalinowski 

This code we had to special case turkish was no longer necessary.  This code used to exist because we would try to normalize down to english's dotted-i form before doing all our pattern matching work.  This pattern matching code would never work on the lower case *non-dotted* turkish i form.

This is no longer necessary because we changed our general completion logic to:

1. first try pattern matching with the current culture's rules.
2. if that didn't work, and the current culture isn't english, try matching with English's rules.

This provides a good system whereby we match properly if the customer is typing identifiers in their own language, while also working when we need to check against things like the framework.

**Comment by rchande** [comment]
:+1:

File: src/EditorFeatures/Core/Extensibility/Completion/CompletionHelper.cs [hunk 1]
```diff
@@ -139,52 +139,37 @@ protected static bool IsAllDigits(string filterText)
...
-        }
-
         private readonly object _gate = new object();
         private readonly Dictionary<string, PatternMatcher> _patternMatcherMap = new Dictionary<string, PatternMatcher>();
         private readonly Dictionary<string, PatternMatcher> _fallbackPatternMatcherMap = new Dictionary<string, PatternMatcher>();
         internal static readonly CultureInfo EnUSCultureInfo = new CultureInfo("en-US");
 
-        protected PatternMatcher GetPatternMatcher(string value, CultureInfo culture)
+        private static PatternMatcher GetPatternMatcher(
+            object gate, string value, CultureInfo culture, Dictionary<string, PatternMatcher> map)
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Extensibility/Completion/CompletionHelper.cs]: Passing the object to lock on seems strange. Consider dropping the `static` and locking on the field again?

**Comment by jasonmalinowski** [comment]
:+1:. The world is a better place.

File: src/EditorFeatures/Core/Extensibility/Completion/CompletionHelper.cs (hunk 1 above)
Comment (DustinCampbell) [inline src/EditorFeatures/Core/Extensibility/Completion/CompletionHelper.cs]: Agreed -- that's super weird. We shouldn't do that.

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by Pilchie** [comment]
Maybe @davkean wants to take a peek at this too?

**Pull request merged**